pkgs/development/rocm-modules/6/aotriton/default.nix

```
@@ -2,226 +2,161 @@
-  hipblas,
-  nlohmann_json,
-  triton-llvm,
-  rocmlir,
-  lapack-reference,
   ninja,
-  ncurses,
-  libffi,
-  zlib,
-  zstd,
   xz,
+  writableTmpDirAsHomeHook,
   pkg-config,
-  buildTests ? false,
-  buildBenchmarks ? false,
-  buildSamples ? false,
-  gpuTargets ? [
+  gpuTargets ? clr.localGpuTargets or clr.gpuTargets,
+  # for passthru.tests
+  aotriton,
+}:
+let
+  supportedTargets = lib.lists.intersectLists [
     # aotriton GPU support list:
     # https://github.com/ROCm/aotriton/blob/main/v2python/gpu_targets.py
     "gfx90a"
     "gfx942"
+    "gfx950"
     "gfx1100"
-    "gfx1101"
-  ],
-}:
-
-stdenv.mkDerivation (
-  finalAttrs:
-  let
-    py = python3.withPackages (ps: [
-      ps.pyyaml
-      ps.distutils
-      ps.setuptools
-      ps.packaging
-      ps.numpy
-      ps.wheel
-      ps.filelock
-      ps.iniconfig
-      ps.pluggy
-      ps.pybind11
-    ]);
-    gpuTargets' = lib.concatStringsSep ";" gpuTargets;
-    compiler = "amdclang++";
-    cFlags = "-O3 -DNDEBUG";
-    cxxFlags = "${cFlags} -Wno-c++11-narrowing";
-    triton-llvm' = triton-llvm;
-  in
-  {
-    pname = "aotriton";
-    version = "0.9.2b";
-
-    src = fetchFromGitHub {
-      owner = "ROCm";
-      repo = "aotriton";
-      rev = "${finalAttrs.version}";
-      hash = "sha256-1Cf0olD3zRg9JESD6s/WaGifm3kfD12VUvjTZHpmGAE=";
-      fetchSubmodules = true;
-    };
-    env.CXX = compiler;
-    env.ROCM_PATH = "${clr}";
-    requiredSystemFeatures = [ "big-parallel" ];
-
-    outputs = [
-      "out"
-    ]
-    ++ lib.optionals buildTests [
-      "test"
-    ]
-    ++ lib.optionals buildBenchmarks [
-      "benchmark"
-    ]
-    ++ lib.optionals buildSamples [
-      "sample"
-    ];
-
-    # Need an empty cuda.h for this to compile
-    # Better than pulling in unfree cuda headers
-    postPatch = ''
-      touch third_party/triton/third_party/nvidia/include/cuda.h
+    "gfx1151"
+    "gfx1150"
+    "gfx1201"
+    "gfx1200"
+  ] gpuTargets;
+  supportedTargets' = lib.concatStringsSep ";" supportedTargets;
+  anySupportedTargets = supportedTargets != [ ];
+  py = python3.withPackages (ps: [
+    ps.wheel
+    ps.distutils
+    ps.setuptools
+    ps.packaging
+    ps.pyyaml
+    ps.numpy
+    ps.filelock
+    ps.iniconfig
+    ps.pluggy
+    ps.pybind11
+    ps.pandas
+    ps.triton
+  ]);
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "aotriton";
+  version = "0.10b";
+
+  src = fetchFromGitHub {
+    owner = "ROCm";
+    repo = "aotriton";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-BvqkCEpDpMe7BBvrHVKUvKol/cE3tujFGoOml+RlHE4=";
+    leaveDotGit = true;
+    # fetch all submodules except unused triton submodule that is ~500MB
+    postFetch = ''
+      cd $out
+      git reset --hard HEAD
+      for submodule in $(git config --file .gitmodules --get-regexp path | awk '{print $2}' | grep '^third_party/' | grep -v '^third_party/triton$'); do
+        git submodule update --init --recursive "$submodule"
+      done
+      rm -rf .git
```

I believe submodules also leave a `.git`
```suggestion
      find "$out" -name .git -print0 | xargs -0 rm -rf
```

remember to refresh the hash